test/compile_test.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+# Run this test from the root of the repository, as:
+# $ bazel test :all --test_output=errors
+
+import unittest
+import subprocess
+import difflib
+import os
+import re
+
+
+class TestSourcesMatchCommittedCSV(unittest.TestCase):
+  """Check that what's committed matches the output of the compiler.
+
+  Any change to this repository that modifies sources for some country's data
+  (the stuff in identifiers/*/*.csv) should also modify the corresponding
+  compiled output (identifiers/country-??.csv), such that they match.
+
+  This test runs the compiler, and compares its output to what's been
+  committed, ensuring they are equal. If this test fails, you need to re-run
+  the compiler for your country, and commit the result.
+  """
+
+  def get_committed_and_compiled_data(self, country_code):
+    # Read what's in the repo.
+    committed_csv_path = F'identifiers/country-{country_code}.csv'
+    with open(committed_csv_path, 'r') as committed_file:
+      committed_csv = committed_file.read()
+
+    # Run the compiler with temp output.
+    compiler_output_path = os.path.join(os.getenv('TEST_TMPDIR'), F"out-{country_code}.csv")
+    subprocess.run([
+        './scripts/compile.py', F'--output_csv={compiler_output_path}', country_code
+    ], check=True)
+
+    # Read the output of the compiler.
+    with open(compiler_output_path, 'r') as compiler_output_file:
+      compiler_output = compiler_output_file.read()
+
+    return committed_csv, compiler_output
+
+  def test_all_countries(self):
+    _, _, master_csv_files = list(os.walk('identifiers'))[0]
+    mismatches = []
+    for filename in master_csv_files:
+      # Ignore any files that don't match the expected pattern.
+      m = re.match(r'country-(..)\.csv', filename)
```

Comment:
I think this will only catch countries with existing country-*.csv. If a new country is added without being compiled, then we would miss it. Could you update this to take into account ones which don't have a csv file compiled already?

Reply:
Currently, if identifiers/country-yy.csv does not exist, but identifiers/country-yy/foo.csv does, the test will ignore yy.

I'll see if I can update the test to prompt the user to run the compiler in that case.

Reply:
Done.